.github/workflows/workflow.yml

```
@@ -83,29 +167,58 @@
+        run: |
+          if [[ "${{ needs.lint-and-typecheck.result }}" == "success" && "${{ needs.build.result }}" == "success" && "${{ needs.e2e-tests.result }}" == "success" ]]; then
+            echo "✅ All critical tests passed!"
+            exit 0
+          else
+            echo "❌ Some tests failed:"
+            echo "Lint & TypeCheck: ${{ needs.lint-and-typecheck.result }}"
+            echo "Build: ${{ needs.build.result }}"
+            echo "Unit Tests: ${{ needs.unit-tests.result }}"
+            echo "E2E Tests: ${{ needs.e2e-tests.result }}"
+            exit 1
+          fi
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)